### ISSUE TITLE
Best Checkpoint Overwriting Due to Incorrect min_loss Update

### ISSUE THREAD

**Issue by sneakybatman** [mindee/doctr#1959]
Hello,

When tracking the best validation loss for model checkpointing, min_loss is updated every epoch, even if the current val_loss is not an improvement. This can lead to overwriting the best checkpoint logic, saving suboptimal models, and misleading progress tracking. min_loss is always set to the last epoch's value, even if it is not actually lower than the true minimum so far. Only update min_loss when a new minimum is found (i.e., when val_loss < min_loss). This ensures that the saved model always corresponds to the best validation loss seen so far. Proposed Fix - Move the [update of min_loss](https://github.com/mindee/doctr/blob/b547085ef4fd661c1a4ba8aebbe77b62c31af9cb/references/recognition/train_pytorch.py#L592) inside the if val_loss < min_loss 

Thanks.

**Comment by felixdittrich92** [comment]
Would you like to open a PR ?

**Comment by sneakybatman** [comment]
Sure will do so.

**Comment by sneakybatman** [comment]
PR is here - https://github.com/mindee/doctr/pull/1961
